source/air/src/context.rs

```diff
@@ -59,7 +60,8 @@ impl<'a, 'b: 'a> Default for QueryContext<'a, 'b> {
     }
 }
 
-pub struct Context {
+pub struct Context<'a> {
```

Comment:
Could we use `Arc<dyn MessageInterface>` or `Rc<dyn MessageInterface>` to avoid the `<'a>` everywhere?